react-popup/my-app/src/Popup.tsx

```diff
@@ -0,0 +1,40 @@
+import { ReactNode } from 'react';
+import { createPortal } from 'react-dom';
+type Props = {
+  isOpen: boolean;
+  onClose: () => void;
+  positionTo?: HTMLElement | null;
+  children: ReactNode;
+};
+
+export function Popup({ isOpen, positionTo, onClose, children }: Props) {
+  const positions = positionTo?.getBoundingClientRect();
+  const top = positions ? positions.top + positions.height : '50%';
+  const left = positions ? positions.left + positions.width / 2 : '50%';
+
+  return createPortal(
+    <>
+      {isOpen && (
+        <div
+          onClick={onClose}
+          style={{
+            position: 'fixed',
+            top: '0',
+            width: '100vw',
+            height: '100vh',
+            filter: 'opacity(50%)',
+            backgroundColor: 'black',
+          }}></div>
+      )}
+      <div
+        className="absolute"
+        style={{
+          top: `${top}`,
```

Comment:
You can do this two different ways:
`style={{ top: top, left: left }}`
or just
`style={{ top, left }}`

These both work because React will convert numbers to a string with `px`.